### PR TITLE
Refactor watchlist updates

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/controllers/person_controller.rb [hunk 1]
```diff
@@ -162,25 +162,14 @@ def internal_register
...
-    remove_from_watchlist.each do |name|
-      user.watched_projects.where(project_id: Project.find_by_name(name).id).delete_all
-    end
-
-    add_to_watchlist.each do |name|
-      user.watched_projects.new(project_id: Project.find_by_name(name).id)
+    new_watchlist.map! do |name|
+      WatchedProject.find_or_create_by(project: Project.find_by_name!(name), user: user)
     end
+    user.watched_projects.replace(new_watchlist)
```
Comment (ChrisBr) [inline src/api/app/controllers/person_controller.rb]: I'm missing that you delete watched projects which got removed from the XML.

Comment (ChrisBr) [inline src/api/app/controllers/person_controller.rb]: I checked the docs, replace deletes projects which are not included! So the implementation looks fine!

See http://www.rubydoc.info/docs/rails/4.1.7/ActiveRecord%2FAssociations%2FCollectionProxy%3Areplace

**Comment by ChrisBr** [comment]
LGTM

File: src/api/app/controllers/person_controller.rb [hunk 2]
```diff
@@ -162,25 +162,14 @@ def internal_register
...
 
     return true
```
Comment (mdeniz) [inline src/api/app/controllers/person_controller.rb]: All LGTM. The only thing is... don't use ```return``` here ;)

**Pull request merged**